CMSgov/bluebutton-web-server
# BB2-291 Create V2 related migration files

**JIRA Ticket:**
[BB2-291](https://jira.cms.gov/browse/BB2-291)

**User Story or Bug Summary:**


This adds migrations for the following model changes:

* Three model changes in the [main V2 PR](https://github.com/CMSgov/bluebutton-web-server/pull/869/files)


### Local migration testing notes:

* Removed BB2 docker images.
  ```
  docker rmi bluebutton-web-server_web  bluebutton-web-server_msls postgres
  ```
* Swiched to "r63" branch
* Start up docker containers:
  ```
  docker-compose up -d
  ```
* Run the migrate.sh script.
* Validated "runtests" is working.
* Perform smoke tests.
* Swiched to "master" branch
* Created a new branch called "dtisza1/BB2-291-Support-BFD-V2-migrations"
* Performed a make migrations: 
  ```
  docker-compose exec web python manage.py makemigrations
  ```
    - This created 7 migrations files. 
      - Cherry picked the migration files related to the V2 PR changes in the following files:
        - apps/accounts/models.py
        - apps/dot_ext/models.py
        - apps/core/models.py
  - The related migration files were:
    - apps/accounts/migrations/0041_auto_20210324_1543.py
    - apps/dot_ext/migrations/0024_auto_20210324_1543.py
    - apps/core/migrations/0002_auto_20210324_1543.py

* Deleted the other migration files.
* Edited the three migration files to JUST include the changes in the PR.
* Performed a show migrations:
  ```
  docker-compose exec web python manage.py showmigrations
  ```
  - Validated that the three migration files are showing with "[ ]" no check mark status.
* Applied the migrations:
  ```
  docker-compose exec web python manage.py migrate
  ```
    - Validate that the three migrations were performed.
* Perform smoke tests.
* Created a V2 migrations PR with these changes.


Other TODOs:





### What Should Reviewers Watch For?




If you're reviewing this PR, please check these things, in particular:


### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then here's a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence: N/A.
    * Does this PR add any new software dependencies? **No**.
    * Does this PR modify or invalidate any of our security controls? **No**.
    * Does this PR store or transmit data that was not stored or transmitted before? **No**.
* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR should not be merged unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons? **No**.


### What Needs to Be Merged and Deployed Before this PR?



This PR cannot be either merged or deployed until the following pre-requisite changes have been fully deployed:


### Submitter Checklist



I have gone through and verified that...:

    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] I have named this PR and its branch such that they'll be automatically be linked to the (most) relevant Jira issue, per: <https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html>.
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. BFD, SLS, etc.).
    * Please review the standards every few months to ensure you're familiar with them.
